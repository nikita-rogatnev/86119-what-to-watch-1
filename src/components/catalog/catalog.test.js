import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";

import Catalog from './catalog';

import mockData from '../../mocks/mock-data';
import mockFilters from '../../mocks/mock-filters';

it(`Catalog renders correctly`, () => {
  const tree = renderer
    .create(<BrowserRouter><Catalog
      data={mockData}
      filters={mockFilters}
      currentFilter={`All genres`}
      changeCurrentFilter={() => {
      }}
      showMoreButton={false}
      showPlayButton={false}
    /></BrowserRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
