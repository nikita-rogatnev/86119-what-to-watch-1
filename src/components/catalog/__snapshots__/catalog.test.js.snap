// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog renders correctly 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Comedies
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Crime
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Documentary
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Dramas
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Horror
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Kids & Family
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Romance
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Sci-Fi
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Thrillers
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Fantastic Beasts: The Crimes of Grindelwald"
          height="175"
          src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Fantastic Beasts: The Crimes of Grindelwald
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Bohemian Rhapsody"
          height="175"
          src="img/bohemian-rhapsody.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Bohemian Rhapsody
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Macbeth"
          height="175"
          src="img/macbeth.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Macbeth
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Aviator"
          height="175"
          src="img/aviator.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Aviator
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="We need to talk about Kevin"
          height="175"
          src="img/we-need-to-talk-about-kevin.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          We need to talk about Kevin
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="What We Do in the Shadows"
          height="175"
          src="img/what-we-do-in-the-shadows.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          What We Do in the Shadows
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Revenant"
          height="175"
          src="img/revenant.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Revenant
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Johnny English"
          height="175"
          src="img/johnny-english.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Johnny English
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Shutter Island"
          height="175"
          src="img/shutter-island.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Shutter Island
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Pulp Fiction"
          height="175"
          src="img/pulp-fiction.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Pulp Fiction
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="No Country for Old Men"
          height="175"
          src="img/no-country-for-old-men.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          No Country for Old Men
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Snatch"
          height="175"
          src="img/snatch.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Snatch
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Moonrise Kingdom"
          height="175"
          src="img/moonrise-kingdom.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Moonrise Kingdom
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Seven Years in Tibet"
          height="175"
          src="img/seven-years-in-tibet.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Seven Years in Tibet
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Midnight Special"
          height="175"
          src="img/midnight-special.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Midnight Special
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="War of the Worlds"
          height="175"
          src="img/war-of-the-worlds.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          War of the Worlds
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Dardjeeling Limited"
          height="175"
          src="img/dardjeeling-limited.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Dardjeeling Limited
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Orlando"
          height="175"
          src="img/orlando.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Orlando
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Mindhunter"
          height="175"
          src="img/mindhunter.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Mindhunter
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Midnight Special"
          height="175"
          src="img/midnight-special.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Midnight Special
        </a>
      </h3>
    </article>
  </div>
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      type="button"
    >
      Show more
    </button>
  </div>
</section>
`;
